c++/src/kj/async-win32.c++

```diff
@@ -158,13 +158,21 @@ Own<Win32EventPort::SignalObserver> Win32IocpEventPort::observeSignalState(HANDL
 }
 
 bool Win32IocpEventPort::wait() {
-  waitIocp(timerImpl.timeoutToNextEvent(clock.now(), MILLISECONDS, INFINITE - 1)
-      .map([](uint64_t t) -> DWORD { return t; })
-      .orDefault(INFINITE));
-
-  timerImpl.advanceTo(clock.now());
-
-  return receivedWake();
+  const bool woke = receivedWake();
+  if(woke) {
+    waitIocp(0);
+    return true;
+  }
```

Comment:
So I thought about it some more, and decided the poll here doesn't actually make sense. `~IoPromiseAdapter` actually did its own poll, after all.

```suggestion
  // It's possible that a wake event was received and discarded during ~IoPromiseAdapter. We
  // need to check for that now.
  if (receivedWake()) return true;
```